test/Google.Api.Gax.Grpc.Tests/ApiCallTest.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using Xunit;
+
+namespace Google.Api.Gax.Grpc.Tests
+{
+    public class ApiCallTest
+    {
+        [Fact]
+        public void WithCallSettingsOverlay()
+        {
+            var ctBase = new CancellationTokenSource().Token;
+            var ctPerCall = new CancellationTokenSource().Token;
+            var ctOverlay = new CancellationTokenSource().Token;
+
+            CallSettings syncCallSettings = null;
+            CallSettings asyncCallSettings = null;
+            var call0 = new ApiCall<SimpleRequest, SimpleResponse>(
+                (req, cs) => { asyncCallSettings = cs; return null; },
+                (req, cs) => { syncCallSettings = cs; return null; },
+                CallSettings.FromCancellationToken(ctBase));
+
+            // Verify a null overlay has no effect.
+            // Async call runs synchonously due to 'call0' definition above.
+            var call1 = call0.WithCallSettingsOverlay(req => null);
+            call1.Sync(null, CallSettings.FromCancellationToken(ctPerCall));
+            call1.Async(null, CallSettings.FromCancellationToken(ctPerCall));
+            Assert.Equal(ctPerCall, syncCallSettings.CancellationToken.Value);
+            Assert.NotEqual(ctBase, syncCallSettings.CancellationToken.Value);
+
+            // Verify an overlay overwrites all else.
```

Comment:
Well, merges with - it's only an overwrite in this case because you're replacing one cancellation token with another. (Maybe that needs testing too?)

Reply:
Correct merging of `CallSettings` is tested elsewhere. This test is just to ensure the merging is done in the correct order.